src/atdf/interrupt.rs

```diff
@@ -5,7 +5,17 @@ use crate::ElementExt;
 pub fn parse(interrupt: &xmltree::Element) -> crate::Result<chip::Interrupt> {
     interrupt.check_name("interrupt")?;
 
-    let name = interrupt.attr("name")?.clone();
+    let name = {
+        let inst_name = interrupt.attr("name")?;
+        interrupt
+            .attr("module-instance")
+            .map_or(inst_name.clone(), |s| {
+                let mut s = s.clone();
+                s.push('_');
+                s.push_str(inst_name);
+                s
+            })
```

Comment:
I suggest using `format!()` instead of manual string manipulation:

```suggestion
            .map_or(inst_name.clone(), |s|
                format!("{}_{}", s, inst_name)
            )
```